ci/generate-and-upload-build-steps.sh

```diff
@@ -2,59 +2,80 @@
-
-        # Linux Development NoEditor build configuration
-        upload_build_configuration_step "${ENGINE_COMMIT_HASH}" "Linux" "" "Development"
+    if [[ -n "${MAC_BUILD:-}" ]]; then
+        if [[ -z "${BUILD_ALL_CONFIGURATIONS+x}" ]]; then
+            # MacOS Development Editor build configuration
+            upload_build_configuration_step "${ENGINE_COMMIT_HASH}" "Mac" "Editor" "Development"
+        else
+            # Editor builds (Test and Shipping build states do not exist for the Editor build target)
+            for BUILD_STATE in "DebugGame" "Development"; do
+                upload_build_configuration_step "${ENGINE_COMMIT_HASH}" "Mac" "Editor" "${BUILD_STATE}"
+            done
+        fi
     else
-        echo "Building for all supported configurations. Generating the appropriate steps..."
-        
-        # Editor builds (Test and Shipping build states do not exist for the Editor build target)
-        for BUILD_STATE in "DebugGame" "Development"; do
-             upload_build_configuration_step "${ENGINE_COMMIT_HASH}" "Win64" "Editor" "${BUILD_STATE}"
-        done
+        # if the BUILD_ALL_CONFIGURATIONS environment variable doesn't exist, then...
+        if [[ -z "${BUILD_ALL_CONFIGURATIONS+x}" ]]; then
+            echo "Building for subset of supported configurations. Generating the appropriate steps..."
+
+            # Win64 Development Editor build configuration
+            upload_build_configuration_step "${ENGINE_COMMIT_HASH}" "Win64" "Editor" "Development"
 
-        # NoEditor, Client and Server builds
-        if [[ "${ENGINE_COMMIT_HASH}" == *"4.22"* ]]; then
-            # Prebuilt engines of native 4.22 and prior do not support Client and Server targets.
-            # We use prebuilt engines in CI, but have manually added two Server configurations:
-            for BUILD_STATE in "Development" "Shipping"; do
-                upload_build_configuration_step "${ENGINE_COMMIT_HASH}" "Linux" "Server" "${BUILD_STATE}"
+            # Linux Development NoEditor build configuration
+            upload_build_configuration_step "${ENGINE_COMMIT_HASH}" "Linux" "" "Development"
+        else
+            echo "Building for all supported configurations. Generating the appropriate steps..."
+
+            # Editor builds (Test and Shipping build states do not exist for the Editor build target)
+            for BUILD_STATE in "DebugGame" "Development"; do
+                upload_build_configuration_step "${ENGINE_COMMIT_HASH}" "Win64" "Editor" "${BUILD_STATE}"
             done
 
-            # NoEditor builds
-            for BUILD_PLATFORM in "Win64" "Linux"; do
-                for BUILD_STATE in "DebugGame" "Development" "Shipping"; do
-                    upload_build_configuration_step "${ENGINE_COMMIT_HASH}" "${BUILD_PLATFORM}" "" "${BUILD_STATE}"
+            # NoEditor, Client and Server builds
+            if [[ "${ENGINE_COMMIT_HASH}" == *"4.22"* ]]; then
```

Comment:
We probably no longer need this, since 4.22 is not supported anymore

Reply:
I was wondering whether I should delete this. Should I?

Reply:
I take the thumbs up from josh as a yes: I'll remove it xD